Library/Formula/rust.rb

```diff
@@ -1,12 +1,34 @@
-require 'formula'
-
 class Rust < Formula
   homepage 'http://www.rust-lang.org/'
-  url 'https://static.rust-lang.org/dist/rustc-1.0.0-beta.3-src.tar.gz'
   version "1.0.0-beta.3"
-  sha256 'e751bc8a8ad236c8865697f866b2863e224af56b0194ddf9f3edd71f9ff6545f'
 
-  head 'https://github.com/rust-lang/rust.git'
+  stable do
+    url 'https://static.rust-lang.org/dist/rustc-1.0.0-beta.3-src.tar.gz'
+    sha256 'e751bc8a8ad236c8865697f866b2863e224af56b0194ddf9f3edd71f9ff6545f'
+
+    resource "cargo" do
+      url "https://github.com/rust-lang/cargo.git", :revision => "83a6d0e", :tag => "0.2.0"
```

Comment:
Needs to be the full revision rather than the short one. Did you test this locally?